### PR TITLE
Hot Chocolate now works like chocolate

### PR DESCRIPTION
**Pull request by super12pl**
## About The Pull Request
Hot chocolate now works like chocolate. (that is mainly felinids reaction)

## Why It's Good For The Game
Hot **Chocolate**

## Changelog
:cl:
add: Hot Chocolate now works like chocolate.
/:cl:

### CODE REVIEW

**Comment by VictorVow** [comment]
Better question is why the hell is `hot_cocoa` a separate reagent.

**Comment by eeSPee** [comment]
> Better question is why the hell is `hot_cocoa` a separate reagent.

It's a drink.

**Comment by Crossedfall** [comment]
Hot_cocoa already has cocoa in it though. 🤔 Is this just extra chocolaty?

https://github.com/BeeStation/BeeStation-Hornet/blob/2467386bb666982dd41cb5e6acb4b1ebc4e392f3/code/modules/food_and_drinks/recipes/food_mixtures.dm#L48-L52

**Comment by super12pl** [comment]
> Hot_cocoa already has cocoa in it though.  Is this just extra chocolaty?
> 
> https://github.com/BeeStation/BeeStation-Hornet/blob/2467386bb666982dd41cb5e6acb4b1ebc4e392f3/code/modules/food_and_drinks/recipes/food_mixtures.dm#L48-L52

You need cocoa and water to create hot cocoa but actual hot cocoa doesn't work like cocoa (felinids can safely drink hot chocolate for example)

**Comment by Crossedfall** [comment]
> 
> You need cocoa and water to create hot cocoa but actual hot cocoa doesn't work like cocoa (felinids can safely drink hot chocolate for example)

So fix hot_cocoa instead 🤔

**Comment by super12pl** [comment]
> > You need cocoa and water to create hot cocoa but actual hot cocoa doesn't work like cocoa (felinids can safely drink hot chocolate for example)
> 
> So fix hot_cocoa instead

Actually yeah I can do that

**Comment by nikk735** [comment]
add candy too its a chocolate bar

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 1]
```diff
@@ -95,7 +95,7 @@
 
 /datum/species/human/felinid/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/M)
 	.=..()
-	if(chem.type == /datum/reagent/consumable/cocoa)
+	if(chem.type == /datum/reagent/consumable/cocoa OR /datum/reagent/consumable/hot_cocoa)
```
Comment (park66665) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]:
```suggestion
	if(istype(chem, /datum/reagent/consumable/cocoa) || istype(chem, /datum/reagent/consumable/hot_cocoa))
```

**Comment by Fronsis** [comment]
As long as it also helps curing the felinid disease... rip the Candy chocolate bar that didn't work as cure

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
yikes repetetive code. Just make hot_cocoa a subtype of cocoa

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 2]
```diff
@@ -95,7 +95,7 @@
 
 /datum/species/human/felinid/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/M)
 	.=..()
-	if(chem.type == /datum/reagent/consumable/cocoa)
+	if(istype(chem, /datum/reagent/consumable/cocoa) || istype(chem, /datum/reagent/consumable/cocoa/hot_cocoa))
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: If you have
A/B/C
and 
A/B/C/D
istype(item, A/B/C) will return true for types A/B/C and A/B/C/D
istype(item, A/B/C/D) will return true for type A/B/C/D

File: code/datums/diseases/transformation.dm [hunk 3]
```diff
@@ -342,7 +342,7 @@
 	cure_text = "Something that would kill off the tiny cats."
 	spread_text = "Acute"
 	disease_flags = CURABLE|CAN_CARRY|CAN_RESIST
-	cures = list(/datum/reagent/consumable/cocoa) //kills all the tiny cats that infected your organism
+	cures = list(/datum/reagent/consumable/cocoa, /datum/reagent/consumable/cocoa/hot_cocoa) //kills all the tiny cats that infected your organism
```
Comment (PowerfulBacon) [inline code/datums/diseases/transformation.dm:345]: Pretty sure subtypes are assumed to work

**Review by PowerfulBacon — COMMENTED** [review]
Also you need to change all instances of /datum/reagent/consumable/hot_cocoa with /datum/reagent/consumable/cocoa/hot_cocoa

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 4]
```diff
@@ -356,7 +356,7 @@
 
 
 
-/datum/reagent/consumable/hot_cocoa
+/datum/reagent/consumable/cocoa/hot_cocoa
 	name = "Hot Chocolate"
 	description = "Made with love! And cocoa beans."
 	nutriment_factor = 3 * REAGENTS_METABOLISM
```
Comment (park66665) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]:
```suggestion
	description = "Made with love! And cocoa beans."
	reagent_state = LIQUID
	nutriment_factor = 3 * REAGENTS_METABOLISM
```

Without this, hot cocoa will be `SOLID`. (afaik it has near-zero gameplay impact)

**Review by PowerfulBacon — DISMISSED** [review]

**Review by park66665 — CHANGES_REQUESTED** [review]
1. typecheck at `/datum/species/human/felinid/handle_chemicals` should be `istype(chem, /datum/reagent/consumable/cocoa)`
2. `/datum/disease` `cures` list is checked by `==` check, not `istype` check. You should change `/datum/disease/transformation/felinid`'s `cures` list to `list(/datum/reagent/consumable/cocoa, /datum/reagent/consumable/cocoa/hot_cocoa)` if you intended `/datum/reagent/consumable/cocoa/hot_cocoa` to work as a cure to `/datum/disease/transformation/felinid`

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**